src/logger/loggerCommands.c

```diff
@@ -93,44 +95,57 @@ void ResetConfig(Serial *serial, unsigned int argc, char **argv)
 
 static void StartTerminalSession(Serial *fromSerial, Serial *toSerial, uint8_t localEcho)
 {
+        bool stay = true;
+        while (stay) {
+                /* Delay so we don't starve other tasks (like watchdog) */
+                delayMs(5);
 
-    while (1) {
-        char c = 0;
-        if (fromSerial->get_c_wait(&c, 0)) {
```

Comment:
behind the scenes it calls this function: http://www.freertos.org/a00118.html.  While it's waiting the task scheduler will be free to work on other things.